### PR TITLE
improving build script

### PR DESCRIPTION
**Pull request by aepfli**
addes some improvements, and i also think fixed one error.

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -3,13 +3,17 @@ name: JUnit Pioneer Site
...
+    types: [triggerSiteBuild]
 
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
       - name: Check out repo
         uses: actions/checkout@v2
+        with:
+          ref: "grandmaster"
```
Comment (aepfli) [inline .github/workflows/actions.yml:16]: i think this action should only run on grandmaster, or am i wrong here, also regarding the build scripts

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -3,13 +3,17 @@ name: JUnit Pioneer Site
 on:
   push:
     branches: [ grandmaster, site-source ]
+  repository_dispatch:
+    types: [triggerSiteBuild]
```
Comment (aepfli) [inline .github/workflows/actions.yml:7]: our trigger from the outside world

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -23,9 +27,7 @@ jobs:
           ruby -v
           bundler -v
           java -version
-      - name: "Gradle: setup"
-        run: ./gradlew setup --no-daemon --stacktrace
-      - name: "Gradle: build site"
-        run: ./gradlew buildSite --no-daemon --stacktrace
-      - name: "Gradle: publish site"
-        run: ./gradlew pushSite --no-daemon --stacktrace
+      - name: Gradle setup, build and push
```
Comment (aepfli) [inline .github/workflows/actions.yml:30]: yes we could use just gradlew, but the gradle command handles all kind of things internally, eg. using gradlew when provided and so on

File: .github/workflows/actions.yml (hunk 1 above)
Comment (Bukama) [reply .github/workflows/actions.yml:16]: > grandmaster: contains all branches (as worktrees) and projects (as clones or symlinks)

So yes, this correct I guess

**Review by Bukama — APPROVED** [review]
Despite the fact I don't know if it works the code itself looks good to me :)

**Pull request merged**